analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -160,34 +141,54 @@ private IEnumerable<string> SortedPackageFolders()
                     : Enumerable.Empty<string>();
             }
 
+            private async Task<string> GetLatestVersion()
+            {
+                var latest = SortedPackageFolders().Select(x => Path.GetFileName(x).Substring(Id.Length + 1)).LastOrDefault(x => char.IsNumber(x[0]));
```

Comment:
This old design is weird. We're looking inside a fragile list to see when it was checked last time. We should have this information available beside the versions.
It doesn't matter on local machines, but on CI there's reused variable NUGET_PACKAGES.
So global cache is on 
NUGET_PACKAGES\packageid\version
and our cache (populated from the global one) is on
NUGET_PACKAGES\packageid.version

Let's change this to
NUGET_PACKAGES\packageid\Sonar.version
NUGET_PACKAGES\packageid\Sonar.NextCheckForUpdate.txt

Like that, we will have a fixed (version independent) location of the timestamp file.